pkgs/development/libraries/science/math/m4ri/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv
+, fetchFromBitbucket
+, autoreconfHook
+}:
+
+stdenv.mkDerivation rec {
+  version = "20140914";
+  name = "m4ri-${version}";
+
+  src = fetchFromBitbucket {
+    owner = "malb";
+    repo = "m4ri";
+    rev = "release-${version}";
+    sha256 = "0xfg6pffbn8r1s0y7bn9b8i55l00d41dkmhrpf7pwk53qa3achd3";
+  };
+
+  doCheck = true;
+
+  nativeBuildInputs = [
+    autoreconfHook
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = http://m4ri.sagemath.org/;
+    description = "Library for matrix multiplication, reduction and inversion over GF(2)";
+    longDescription = ''
+      M4RIE is a library for fast arithmetic with dense matrices over small finite fields of even characteristic. It uses the M4RI library, implementing the same operations over the finite field F2.
+    '';
+    license = with licenses; [ gpl2 ];
+    maintainers = with maintainers; [ timokau ];
+    platforms = platforms.all;
```

Comment:
Have you tested all platforms? It's generally better to set to Linux only or to `platforms.unix` if Darwin was also tested.

Reply:
No, but I know that it works on various platforms in the sage distribution (including ARM, darwin, cygwin) so I have reasonable hope that it'll work.

Reply:
Please set `platforms = platforms.linux` until someone builds this on macOS.